### DISCUSSION TITLE
Should IFrameGrabber and IFrameGrabberRgb be deprecated?

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/yarp#1868] [Architecture]
The `IFrameGrabber` and the `IFrameGrabberRgb` interfaces seem to have been superseded by the `IFrameGrabberImage` and the `IFrameGrabberImageRaw` interfaces since a long time, they are not used in any network driver (not even for back-compatibility) and can be found only in some devices in  `icub-main` ( https://github.com/robotology/icub-main/search?q=IFrameGrabberRgb&unscoped_q=IFrameGrabberRgb), but all those devices  also offer a variant that implements the modern interfaces `IFrameGrabberImage` and the `IFrameGrabberImageRaw` . 

I am not an expert of this part of YARP, but I guess this interfaces can be safely deprecated and eventually removed from YARP.

@vtikha @barbalberto @drdanz @Nicogene @claudiofantacci

**Comment by Nicogene** [comment]
I found out that `IFrameGrabberRgb` and `IFrameGrabber` are used in `dragonfly2` and `usbCamera` (`icub-main` devices).

@barbalberto , are they easily portables to `IFrameGrabberImage` and `IFrameGrabberImageRaw`?

If so we can deprecate the old ones as @traversaro suggested.

**Comment by traversaro** [comment]
`dragonfly2` and `usbCamera` actually come in two versions, one using the old interfaces and another using the new one. It is a bit tricky to understand this because both devices are in the same source code file.

**Comment by barbalberto** [comment]
In my opinion they can be deprecated.
The only update required is in the [usbCamera](https://github.com/robotology/icub-main/blob/c3ad642ed0aacdb3216f692cc73916936079714b/src/libraries/icubmod/usbCamera/common/USBcamera.cpp#L382) where the new interface is calling the old one of the lower level implementation.

It should propagate the new one and the low level device shall implement it [here ](https://github.com/robotology/icub-main/blob/c3ad642ed0aacdb3216f692cc73916936079714b/src/libraries/icubmod/usbCamera/linux/V4L_camera.cpp#L922)

Few changes are required, not a big deal.

**Comment by drdanz** [comment]
I made a quick test commenting out these interfaces, and I can confirm that they are completely unused in YARP.

**Comment by drdanz** [comment]
usbCamera is now in YARP, therefore we should update it before removing the interface
